### ISSUE TITLE
0.62.0 Discussion

### ISSUE THREAD

**Issue by kelset** [react-native-community/releases#179]
Conversations on this thread are limited to 0.62 releases major issues and backport (cherry-pick) requests from commits that are already on master.

An example of a good such request is a bug fix for a serious issue that has been merged into master but did not make the 0.62.0 cut.

In other words, if you cannot point to a particular commit on master, then your request likely belongs as a new issue in http://github.com/facebook/react-native/issues.

---

We have released this version to respect the work of hundreds of contributors who made this release possible and to prevent the release from falling too far behind master. Please be mindful of the reduced capacity of contributors to help with issues and prepare to delay upgrading if necessary.

**Comment by HarshitMadhav** [comment]
From all of us! **Heartily Thanks** to all the contributors who made this release possible in the time of Global Pandemic. Also, by this comment I would request all to **'Stay At Home'** and follow their Government Policies.

**Comment by milo-** [comment]
Thank you for the amazing work getting this out in such difficult times. Would it be possible to get https://github.com/facebook/react-native/pull/27574 backported into 0.62.x?

**Comment by jeveloper** [comment]
From the bottom of my heart (high and vulnerable) i want to wish all of you to be safe, healthy and close with other fellow humans. We're in this together across the world. Thank you and lots of love. The world will be bright again, more tollerant and accepting!

**Comment by farwayer** [comment]
Hi guys! We need to cherry-pick this commit https://github.com/facebook/react-native/commit/cfcf5eba4317f80ef8902463b7c0b2e1e7b534a7 to prevent crash in release build on Android with hermes engine and `enableProguardInReleaseBuilds` enabled.

```java.lang.UnsatisfiedLinkError: couldn't find DSO to load: libhermes.so caused by: Didn't find class "com.facebook.jni.NativeRunnable" on path: DexPathList[[zip file "/data/app/com.alormi-0LUU5G6NefGa4kK5cebJ3A==/base.apk"],nativeLibraryDirectories=[/data/app/com.alormi-0LUU5G6NefGa4kK5cebJ3A==/lib/arm, /data/app/com.alormi-0LUU5G6NefGa4kK5cebJ3A==/base.apk!/lib/armeabi-v7a, /system/lib]]```

**Comment by kesha-antonov** [comment]
Flipper doesn't build on ios
Works fine on android

**Comment by sunnylqm** [comment]
Need to follow flipper doc https://fbflipper.com/docs/getting-started.html#cocoapods  @kesha-antonov

**Comment by kesha-antonov** [comment]
@sunnylqm Cool! Thanks
I'll try that

**Comment by kesha-antonov** [comment]
Didn't help...

**Comment by ahce** [comment]
Hi! please pick this commit facebook/react-native@d3658bc2b6437e858d3b3f5688277dedbca779b8
Thanks!

**Comment by soroushm** [comment]
react-native run-ios is faild

already try `rm -rv ~/.rncache`
```

warning: no rule to process file '/Users/m.soroush/Sites/mobiletrader/ios/Pods/Flipper-RSocket/rsocket/benchmarks/CMakeLists.txt' of type 'text' for architecture 'x86_64' (in target 'Flipper-RSocket' from project 'Pods')
warning: no rule to process file '/Users/m.soroush/Sites/mobiletrader/ios/Pods/Flipper-RSocket/rsocket/benchmarks/README.md' of type 'net.daringfireball.markdown' for architecture 'x86_64' (in target 'Flipper-RSocket' from project 'Pods')

** BUILD FAILED **


The following build commands failed:
        CompileC /Users/m.soroush/Library/Developer/Xcode/DerivedData/Farabixo-eiylqursbwqzuzbrkgeuejfkwdbk/Build/Intermediates.noindex/Farabixo.build/Debug-iphonesimulator/Farabixo.build/Objects-normal/x86_64/AppDelegate.o /Users/m.soroush/Sites/mobiletrader/ios/Farabixo/AppDelegate.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

**Comment by kelset** [comment]
@kesha-antonov @soroushm please open issues in the main repo to report your problems (first ofc check if someone already reported them), and link them back here.

Also, if you think your issue may be related to upgrading, please refer to the new [Upgrade Support](https://github.com/react-native-community/upgrade-support) repository.

**Comment by soroushm** [comment]
@kelset I upgraded manually `upgrade-support` completely failed from 61.5 to 62
looks like a  linting error or something's like this

**Comment by marcorm** [comment]
Hi, I'm trying to update a library to get rid of this warning:
```
Calling getNode() on the ref of an Animated component is no longer necessary
```
Changing
`this.contentHandle.getNode().measure(...)`
to
`this.contentHandle.measure(...)`
works on 0.62.0.

My question is: **is it safe to do so? Won't I brake integration with older RN versions?**

I hope it's the right place to post this question, since it's related to a 0.62.0.
Thank you!

**Comment by TheSavior** [comment]
Can we pick `https://github.com/facebook/react-native/commit/405200e9a930cded47954f374f2a779ec769cd4c`

**Comment by TheSavior** [comment]
@marcorm, let's please keep this issue focused on the commits that need to get picked for a patch release. Please open an issue on the main repo with your question. That said, you could check for the existence of measure on `this.contentHandle` and call it, otherwise call `getNode`.

**Comment by heroic** [comment]
@farwayer you are a savior!

**Comment by kelset** [comment]
to cherry pick too: https://github.com/facebook/react-native/commit/227aa96bb23b6ff20eebbd8a9335fd172ed6005b

**Comment by xiao99xiao** [comment]
Just realized several UIScene related PRs didn't make into 0.62, could we get them cherry picked?
https://github.com/facebook/react-native/pull/25425
https://github.com/facebook/react-native/pull/28141
https://github.com/facebook/react-native/pull/28147

**Comment by gorhom** [comment]
Hi 👋
could we pick https://github.com/facebook/react-native/commit/576ddfb3a84a5461679959f0d3f229a000dcea8d

**Comment by xiao99xiao** [comment]
This commit https://github.com/facebook/react-native/commit/a5353c0361f330b2b8aebc5de8dc2d006887081b , I don't why it's included in 0.61 but not in 0.62. Maybe because it's not added by PR?

Any way this commit is very important as it fixed this Modal issue https://github.com/facebook/react-native/issues/26473

@kelset  Could we cherry pick the commit and make sure it will not be absent again in the next big version like 0.63 or something?

**Comment by alloy** [comment]
@xiao99xiao Those UIScene changes are not bug-fixes, so they won’t go into a patch release. They will be available in v0.63.0, though!

**Comment by timurgaitov** [comment]
If someone encounters errors while building the ios app, try updating cocoapods.

**Comment by SaeedZhiany** [comment]
@xiao99xiao , @alloy 

I need to know before going to upgrade my project, Modal's onDismiss functionality doesn't work properly right now?

**Comment by xiao99xiao** [comment]
> @xiao99xiao , @alloy
> 
> I need to know before going to upgrade my project, Modal's onDismiss functionality doesn't work properly right now? (I'm currently using RN 0.59.10)

Yes, it's not working in 0.62. But you are safe to go for 0.61, or wait for 0.62.1.

Let's bring the discussion to the right place https://github.com/facebook/react-native/issues/26473 . This thread is for backport support track only.

**Comment by kelset** [comment]
hey folks, quick update, we will try to get a 0.62.1 release out later today. For this patch release we will (likely*):

* cherry pick https://github.com/facebook/react-native/commit/7a4753d76aab1c52a09f26ec6f7fd43a68da8a97
* cherry pick https://github.com/facebook/react-native/commit/cfcf5eba4317f80ef8902463b7c0b2e1e7b534a7
* cherry pick https://github.com/facebook/react-native/commit/d3658bc2b6437e858d3b3f5688277dedbca779b8
* cheery pick https://github.com/facebook/react-native/commit/405200e9a930cded47954f374f2a779ec769cd4c
* won't cherry pick this https://github.com/facebook/react-native/commit/227aa96bb23b6ff20eebbd8a9335fd172ed6005b because it's been done directly on 0.62 branch hence it's already there :D
* cherry pick https://github.com/facebook/react-native/commit/576ddfb3a84a5461679959f0d3f229a000dcea8d
* to fix the modal support as reported by @xiao99xiao we will likely replicate this PR as a local commit https://github.com/facebook/react-native/pull/26490 (instead of doing the revert)
* bump RN CLI minor to the latest available version (4.5.1 at the time of writing) so that people get updated CLI without mangling with lock files


\* as always, it depends on the cherry picking process & our local testing to go well 🤞

**Comment by xiao99xiao** [comment]
@kelset However this PR https://github.com/facebook/react-native/pull/26490 is not a valid fix compared with the revert.

**Comment by alloy** [comment]
> @kelset However this PR [facebook/react-native#26490](https://github.com/facebook/react-native/pull/26490) is not a valid fix compared with the revert.

@xiao99xiao Can you please elaborate on exactly why not and what you’d suggest we should be doing instead?

**Comment by xiao99xiao** [comment]
@alloy OK. The key difference about the revert commit and the PR's solution is, the definition about dismiss on iOS.

On iOS, when we say a Modal is dismissed, it means the view is not visible **and its view controller is removed from the parent view controller**. And another important thing is, you could only present a new Modal **after the earlier one's view controller is removed too**.

The PR only watches on the visibility of the Modal view. However when the Modal view is not visible, its view controller could be not removed yet. Merging the PR will break one common logic that you need to present another Modal after one's dismissed.

So that's why I insist using the revert: The PR and the former commit are not right about dismiss and break too much.

**Comment by kelset** [comment]
Hey @xiao99xiao thank you a lot for your explanation 🙏

We've reached out to the author of the PR and he agrees with your assesment, so we will consider an alternative approach to solve the Modal issue (probably we will do the revert as you suggested in the first place).

I'll write an update here once we have done what above

**Comment by gorhom** [comment]
also could we pick https://github.com/facebook/react-native/pull/28512, it fixes RCTPerfMonitor crashes on iOS 13.4

**Comment by alloy** [comment]
Hey everybody, we just released v0.62.1: https://github.com/facebook/react-native/releases/tag/v0.62.1 🚀 

You can see in @kelset’s comment what changes we applied or go directly to the [CHANGELOG](https://github.com/react-native-community/releases/blob/master/CHANGELOG.md#v0621). Regarding the `Model` `onDismiss` prop we reverted the incorrect fix to be on par with v0.61.x.

As for other changes that were requested but _not_ yet landed on `master`, we did not pick those as that’s not our policy 🙏 

Thanks for all of your assistance and looking forward to continuing our collaborations on [the v0.62.1 issue](https://github.com/react-native-community/releases/issues/184)!
